infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -63,13 +69,15 @@ public LogEntryWriter(LogReplicationMetadataManager metadataManager,
         this.session = session;
 
         if (session.getSubscriber().getModel() == LogReplication.ReplicationModel.ROUTING_QUEUES) {
-            String replicatedQueueName = ((LogReplicationRoutingQueueConfig) replicationContext.getConfig(session))
-                    .getSinkQueueName();
-            replicatedRoutingQueueStreamId = CorfuRuntime.getStreamID(replicatedQueueName);
+            replicatedRoutingQUuid = CorfuRuntime.getStreamID(
+                    TableRegistry.getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE,
+                            REPLICATED_RECV_Q_PREFIX + session.getSourceClusterId() + "_" +
+                                    session.getSubscriber().getClientName())
+            );
             replicatedRoutingQueueStreamTag = ((LogReplicationRoutingQueueConfig) replicationContext.getConfig(session))
                     .getSinkQueueStreamTag();
         } else {
-            replicatedRoutingQueueStreamId = null;
+            replicatedRoutingQUuid = null;
```

Comment:
would be great to use Optional for the variable

Reply:
This variable would not be used at all once set to null...we won't even have a null check.
Instead we check on the replication_model.

So I think its okay to use null here. Please let me know if you still think that using optional would be good here.